src/aggregate_score.rs

```diff
@@ -1,16 +1,18 @@
-use std::fmt;
-use rand::Rng;
-
+use crate::hand::Hand;
 use crate::hand_score::display_hand_data;
 use crate::hand_score::HandData;
 use crate::hand_score::HandScore;
-use crate::hand::Hand;
 use crate::hand_stats::HandStats;
+use rand::rngs::ThreadRng;
+use rand::Rng;
+use rayon::prelude::*;
+use std::fmt;
+use std::sync::Arc;
+use std::sync::Mutex;
 
 pub type AggregateScore = HandData<u32>;
 
 impl AggregateScore {
-
     #[allow(dead_code)]
     pub fn insert(&mut self, score: &HandScore) {
         self.high_card += 1;
```

Comment:
How do I add generic here, I couldn't find any trait that allows only primitive types or that supports bool + numeric. Maybe I am missing something. You have some suggestions?

Reply:
Hmmmm... this might be harder than I had though. I'm still used to C++ template land where this would be a simple thing (because there is less structure around bool to int conversions).

I'm away from my computer for a bit, so I can't hack on it right now, although perhaps there is a solution.

For now, I'm fine dropping this suggestion from the scope of your PR and just leaving the near-duplicate code.

Reply:
Another option here might be to just drop the `bool` version of this entirely, and just use the `u32` version everywhere. That would simplify the codebase and work basically the same. The debug printing wouldn't be as nice, but that's ok. Feel free to do that here, or I can do it later in a follow-up PR.

Reply:
So should I treat true and false as 1 and 0 ?

Reply:
Yep!